common/dynamicconfig/collection.go

```diff
@@ -308,16 +308,24 @@ func subscribe[T any](
 	c.subscriptionLock.Lock()
 	defer c.subscriptionLock.Unlock()
 
+	// get one value immediately (note that subscriptionLock is held here so we can't race with
+	// an update)
+	init := matchAndConvert(c, key, def, cdef, convert, prec)
+	
+	// As a convenience (and for efficiency), you can pass in a nil callback; we just return the
+	// current value and skip the subscription.  The cancellation func returned is a no-op and
+	// can be called or ignored.
+	if reflect.ValueOf(callback).IsNil() {
```

Comment:
Why not `callback == nil`? It's a concrete type here, not an interface type

Reply:
a func argument is by definition an interface type

Reply:
As far as I know this isn't true, there's nothing here that suggests a function has anything to do with interfaces: https://go.dev/ref/spec#Function_types

To put it concretely, how could I construct a call such that `callback == nil` is false but `reflect.ValueOf(callback).IsNil()` is true?